addons/account_asset/account_asset.py

```diff
@@ -1,351 +1,325 @@
+            raise ValidationError(_('Prorata temporis can be applied only for time method "number of depreciations".'))
+
+    @api.onchange('category_id')
+    def onchange_category_id(self):
+        vals = self.onchange_category_id_values(self.category_id.id)
+        # We cannot use 'write' on an object that doesn't exist yet
+        if vals:
+            for k, v in vals['value'].iteritems():
+                setattr(self, k, v)
+
+    def onchange_category_id_values(self, category_id):
         if category_id:
-            category_obj = asset_categ_obj.browse(cr, uid, category_id, context=context)
-            res['value'] = {
-                            'method': category_obj.method,
-                            'method_number': category_obj.method_number,
-                            'method_time': category_obj.method_time,
-                            'method_period': category_obj.method_period,
-                            'method_progress_factor': category_obj.method_progress_factor,
-                            'method_end': category_obj.method_end,
-                            'prorata': category_obj.prorata,
+            category = self.env['account.asset.category'].browse(category_id)
+            return {
+                'value': {
+                    'method': category.method,
+                    'method_number': category.method_number,
+                    'method_time': category.method_time,
+                    'method_period': category.method_period,
+                    'method_progress_factor': category.method_progress_factor,
+                    'method_end': category.method_end,
+                    'prorata': category.prorata,
+                }
             }
-        return res
 
-    def onchange_method_time(self, cr, uid, ids, method_time='number', context=None):
-        res = {'value': {}}
-        if method_time != 'number':
-            res['value'] = {'prorata': False}
+    @api.onchange('method_time')
+    def onchange_method_time(self):
+        if self.method_time != 'number':
+            self.prorata = False
+
+    @api.multi
+    def copy_data(self, default=None):
+        if default is None:
+            default = {}
+        default.update({'name': self.name + _(' (copy)')})
+        return super(account_asset_asset, self).copy_data(default)[0]
+
+    @api.multi
+    def _compute_entries(self, period_id):
+        period = self.env['account.period'].browse(period_id)
+        depreciation_ids = self.env['account.asset.depreciation.line'].search([
+            ('asset_id', 'in', self.ids), ('depreciation_date', '<=', period.date_stop),
+            ('depreciation_date', '>=', period.date_start), ('move_check', '=', False)])
+        self = self.with_context(depreciation_date=period.date_stop)
+        return depreciation_ids.create_move()
+
+    @api.model
+    def create(self, vals):
+        self = self.with_context(mail_create_nolog=True)
```

Comment:
No, If you want to invoke a method with a specific context, make a recordset(s) with that context and invoke the method on that one.